agent/engine/dependencygraph/graph.go

```diff
@@ -367,6 +372,15 @@ func containerOrderingDependenciesIsResolved(target *apicontainer.Container,
 	}
 }
 
+func hasDependencyTimedOut(dependOnContainer *apicontainer.Container, dependencyCondition string) bool {
+	if dependencyCondition == successCondition || dependencyCondition == completeCondition || dependencyCondition == healthyCondition {
```

Comment:
Couple things:

* I feel like this should return "true" when there is a timeout.

* I'd rather use a switch:
```
switch  dependencyCondition {
case success,complete,healthy:
   // it _has_ timed out if this returns true
    return time.Now().Af....
default:
   // it has not timed out
    return false
}
```